.github/workflows/ci.yml

```diff
@@ -249,22 +249,33 @@ jobs:
       - uses: rui314/setup-mold@v1
       - name: wasm target install
         run: rustup target add wasm32-unknown-unknown
+      - name: Install cargo-nextest
+        run: cargo install cargo-nextest --locked
       - name: cargo test compile
         uses: actions-rs/cargo@v1
         with:
           command: test
           args: --no-run --locked --all-features --release
+
       - name: cargo test
+        run: cargo nextest run --all-features --release -E "not package(integration_tests)" --profile ci
+      - name: upload artifact
+        uses: actions/upload-artifact@v3  # upload test results as artifact
+        if: always()
```

Comment:
You want to upload on cancels as well I gather?

Reply:
ah no, this should be success() || failure() as well, thanks

Reply:
Not 100% sure, but `always()` seems to change the outcome to success if the previous step failed